File: aspect/src/main/java/com/omidmohebbise/aspect/domain/Person.java
package com.omidmohebbise.aspect.domain;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Table(name = "persons")
@Getter
@Setter
@NoArgsConstructor
public class Person {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;
    public String name;
    public String family;


    public static Person factory(String name, String family) {
        Person p = new Person();
        p.setName(name);
        p.setFamily(family);
        return p;
    }

}

File: kotlin/src/main/kotlin/com/omidmohebbise/kotlin/domain/Person.kt
package com.omidmohebbise.kotlin.domain

import javax.persistence.*

@Table(name = "persons")
@Entity
class Person {
    @Id @GeneratedValue var id: Long? = null
    var fullName: String = ""


    constructor(fullName: String) {
        this.fullName = fullName
    }
    constructor(){

    }
}